### DISCUSSION TITLE
Adding papers from the API

### DISCUSSION THREAD

**Discussion by kiike** [papis/papis#765] [Ideas]
I recently tried to submit a PR for papis-zotero to use papis.yaml to save the yaml file, but I thought of a perhaps better way to import files.

Right now, papis-zotero creates the yaml and the directory structure missing features such as formatted file or folder names, or not handling duplicates, which would handle papis/papis-zotero#35, for instance.

I am [prototyping](https://github.com/papis/papis-zotero/compare/main...kiike:papis-zotero:pr-use-papis-add-for-import) to use `papis.commands.add.run `to add the files in my personal fork, getting the attachments from the database too.

`papis.commands.add.run` already implements the logic to handle duplicates, checking if a document is already in the library and asking for confirmation. It also tells you where it is placing the files and also applies the user's config to handle adding files.

So, I thought what about using it for imports? And it actually works great!

However, I am not sure if using the `papis.commands.add.run` is clean at all. What do you think? I'd be happy to submit a PR.

**Comment by alexfikl** [comment]
Yes, that's definitely a good idea! I think the `bibtex` importer in `papis-zotero` already uses `papis.commands.add.run`, so we should unify those.

A PR with that would be lovely :grin:

**Comment by kiike** [comment]
Hey @alexfikl! Thanks for the reply! Awesome! It didn't occur to me to have a look at the bibtex importer :stuck_out_tongue:.

I'll check what that importer is doing and submit a PR afterwards.
